test/chttpd_httpd_handlers_tests.erl

```diff
@@ -0,0 +1,54 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License.  You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(chttpd_httpd_handlers_tests).
+
+-export([handlers/1]).
+
+-include_lib("couch/include/couch_eunit.hrl").
+
+handlers(url_handler) ->
+    [
+        {<<"">>, chttpd_misc, handle_welcome_req},
+        {<<"favicon.ico">>, chttpd_misc, handle_favicon_req},
+        {<<"_utils">>, chttpd_misc, handle_utils_dir_req},
+        {<<"_all_dbs">>, chttpd_misc, handle_all_dbs_req},
+        {<<"_active_tasks">>, chttpd_misc, handle_task_status_req},
+        {<<"_node">>, chttpd_misc, handle_node_req},
+        {<<"_reload_query_servers">>, chttpd_misc, handle_reload_query_servers_req},
+        {<<"_replicate">>, chttpd_misc, handle_replicate_req},
+        {<<"_uuids">>, chttpd_misc, handle_uuids_req},
+        {<<"_session">>, chttpd_auth, handle_session_req},
+        {<<"_up">>, chttpd_misc, handle_up_req},
+        {<<"anything">>, chttpd_db, handle_request}
```

Comment:
> I am also thinking rather it would be better to place tests in <app>_httpd_handlers.erl itself.

+1 - this will at least help to not miss the tests if you're going to modify handlers. My point here was about to not fall into situation, when you change/update `<app>_httpd_handlers.erl` and not update the tests (they may eventually stay green after) and imho this should help a bit.

> It would be trivial with elixir macro :-).

I think use elixir for testing is not a bad idea at all. They have quite friendly and powerful tools for that. But not today (: